src/test/java/guitests/LabelPickerTests.java

```diff
@@ -56,6 +57,9 @@ public void addAndRemoveLabelTest() {
         sleep(EVENT_DELAY);
         assertEquals(0, listPanelCell.getIssueLabels().size());
 
+        // sleep long enough for label changes to be made permanent (i.e. longer than the undo window period)
+        sleep(Notification.DEFAULT_NOTIFICATION_PANE_VISIBLE_PERIOD * 1000);
```

Comment:
Could this be synchronised using `UITest.waitUntilNodeDisappears`? As in waiting for the notification UI to go away.

Also, could you see if the tests work with `waitOnFxThread`?